src/cpp/session/SessionConsoleInput.cpp

```diff
@@ -362,7 +362,8 @@ bool rConsoleRead(const std::string& prompt,
    if (!pConsoleInput->isCancel())
    {
       module_context::events().onBeforeExecute();
-      module_context::events().onConsoleInput(pConsoleInput->text);
+      if (!pConsoleInput->isNoEcho())
```

Comment:
I'm not sure if we want to suppress these events here -- RStudio modules might still want to take action in response to console input even if we're not echoing it to the user.

Reply:
Thanks.